### ISSUE TITLE
Jackett NAS lost all its configuration

### ISSUE THREAD

**Issue by almico** [alexbelgium/hassio-addons#135]
All of a sudden, one hour ago, all indexers disappeared. Even the API KEY changed.
This is what I see in the logs. HA didn't restart. Only Jackett NAS, somehow, restarted.

```
[cont-init.d] 00-aaa_dockerfile_backup.sh: executing... 
[cont-init.d] 00-aaa_dockerfile_backup.sh: exited 0.
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Add-on: Jackett NAS
 translates queries from apps (Sonarr, SickRage, CouchPotato, Mylar, etc) into tracker-site-specific http queries, parses the html response, then sends results back to the requesting software
-----------------------------------------------------------
 Add-on version: 0.20.216
 You are running the latest version of this add-on.
 System: Home Assistant OS 7.0  (aarch64 / raspberrypi4-64)
 Home Assistant Core: 2021.12.7
 Home Assistant Supervisor: 2021.12.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums
 https://github.com/alexbelgium/hassio-addons
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-envfile: executing... 
[cont-init.d] 01-envfile: exited 0.
[cont-init.d] 01-migrations: executing... 
[migrations] started
[migrations] no migrations found
[cont-init.d] 01-migrations: exited 0.
[cont-init.d] 10-adduser: executing... 
-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \ 
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/
Brought to you by linuxserver.io
-------------------------------------
To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------
User uid:    0
User gid:    0
-------------------------------------
[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 20-folders.sh: executing... 
Moving to new location /config/addons_config/Jackett
[cont-init.d] 20-folders.sh: exited 0.
[cont-init.d] 30-config: executing... 
[cont-init.d] 30-config: exited 0.
[cont-init.d] 90-custom-folders: executing... 
[cont-init.d] 90-custom-folders: exited 0.
[cont-init.d] 92-local_mounts.sh: executing... 
[cont-init.d] 92-local_mounts.sh: exited 0.
[cont-init.d] 92-smb_mounts.sh: executing... 
[cont-init.d] 92-smb_mounts.sh: exited 0.
[cont-init.d] 99-custom-scripts: executing... 
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-scripts: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
01-01 08:18:47 Info Starting Jackett v0.20.222 
01-01 08:18:49 Info Environment version: 6.0.1 (/app/Jackett/) 
01-01 08:18:49 Info OS version: Unix 5.10.63.8 (64bit OS) (64bit process) 
01-01 08:18:49 Info Jackett variant: CoreLinuxArm64 
01-01 08:18:49 Info File /etc/issue: Ubuntu 18.04.6 LTS \n \l 
01-01 08:18:49 Info Running in Docker: Yes (image build: v0.20.222-ls54) 
01-01 08:18:49 Info ThreadPool MaxThreads: 32767 workerThreads, 1000 completionPortThreads 
01-01 08:18:49 Info App config/log directory: /config/Jackett 
01-01 08:18:49 Info Using proxy: Disabled 
01-01 08:18:49 Info Using FlareSolverr: No 
01-01 08:18:49 Error Jackett is running with root privileges. You should run Jackett as an unprivileged user. 
01-01 08:18:49 Info Using HTTP Client: HttpWebClient2 
01-01 08:18:49 Info Loading Native indexers ... 
01-01 08:18:50 Info Loaded 87 Native indexers: (removed)
01-01 08:18:50 Info Loading Cardigann indexers from: /config/cardigann/definitions/, /etc/xdg/cardigan/definitions/, /app/Jackett/Definitions 
01-01 08:18:57 Info Loaded 497 Cardigann indexers: (removed)
01-01 08:18:57 Info Loaded 584 indexers in total 
01-01 08:18:57 Info Adding aggregate indexer ('all' indexer) ... 
01-01 08:18:58 Info Adding filter indexer ('type:public' indexer) ... 
01-01 08:18:58 Info Adding filter indexer ('type:private' indexer) ... 
01-01 08:18:58 Info Adding filter indexer ('type:semi-public' indexer) ... 
01-01 08:18:58 Warn No XML encryptor configured. Key {(removed)} may be persisted to storage in unencrypted form. 
01-01 08:18:58 Warn No XML encryptor configured. Key {(removed)} may be persisted to storage in unencrypted form. 
01-01 08:18:58 Info Jackett startup finished in 12.764 s 
Hosting environment: Production
Content root path: /app/Jackett/Content
Now listening on: http://[::]:9117
Application started. Press Ctrl+C to shut down.
```

**Comment by almico** [comment]
I restored a backup (version 0.20.197) and, unfortunately, all indexers are gone and a new API KEY was generated.
The backup was, previously, working perfectly.

**Comment by alexbelgium** [comment]
Thanks, I think I think it's due to changing the config location. If you restore your addon, you need to move the folder in /config/addons_config/Jackett to /config/Jackett and you'll get back your config. I'll investigate, thanks

Btw: happy new year! 

@bruvv : hi, looks like we overlooked something in the config location change. Is it working for you?

**Comment by alexbelgium** [comment]
I think i found the root cause. 

Please update it should work again, without needing to move the config folder at its original location 

If you lost your data still, you'll need to restore the config data in addition to the addon, that's where the info is stored... But as the config folder was not deleter, merely moved, it should work!

**Comment by alexbelgium** [comment]
BTW, delete the new config folder before updating or restarting! Or it will probably erase your old config, and you'll have to restore your config folder from a backup to restore your Jackett config

**Comment by almico** [comment]
I just noticed that in config directory there are two:

```
Jackett
jackett
```

**Comment by alexbelgium** [comment]
BTW, rename the new config folder in /config/Jackett to Jackett.bak for example before updating or restarting! Or it will probably erase your old config, and you'll have to restore your config folder from a backup to restore your Jackett config

**Comment by alexbelgium** [comment]
> I just noticed that in config directory there are two:
> 
> ```
> Jackett
> jackett
> ```

The correct one is Jackett with the capital J, the one with jackett is normally empty. We corrected that with bruvv a few days ago to avoid its creation, but as it, as already created its there but safe to delete (if truly empty)

**Comment by alexbelgium** [comment]
Thanks for reporting the issue :) we have moved with bruvv all addons config to /config/addons_config to avoid crowding /config with too many folders, but we couldn't test everything to confirm it worked...

**Comment by almico** [comment]
Hmmm... I need to properly understand which is the correct directory and WHEN it will contain the right data after a restore 😄

**Comment by alexbelgium** [comment]
Sorry, I'll explain a bit more.

Previously, there were two folders : /config/Jackett (contained Jackett config files) and /config/jackett (empty).

With @bruvv, we standardized all addons config files in /config/addons_config. To do that, the addon migrates on first launch the content of /config/Jackett to /config/addons_config/Jackett, then deletes /config/Jackett.

However, there was an error in the code. Therefore, even though your old config was saved in /config/addons_config /Jackett, the addon was still looking in /config/Jackett and created a new config folder with brand new values.

So, to summarize:
- /config/jackett is empty and can be deleted
- /config/Jackett contains the new config and can be deleted
- if you only started the addon once, /config/addons_config/Jackett contains your old config. If you update to the new version I've just pushed (thar finishes by -2), it will use this folder all right. If you started the addon twice, it will have erased your old config with the new one. Therefore, you'll need to restore your /config folder from a backup to recover the /config/Jackett folder that contains your files...

Hope I was clearer this time :)

**Comment by bruvv** [comment]
Happy new year :) I was already working on a fix that you merged which appeared to be empty... Will look into it again why that didn't work. Sorry! Will fix it with the next update.

**Comment by almico** [comment]
Ok! Looks like it's alive again. I, wrongly, expected the configuration files to be in the Jackett NAS backup itself. I had to manually extract the relevant configuration files from a HA Core backup instead.
I also had to refresh the addons store to have it know about the latest update 😄

**Comment by alexbelgium** [comment]
Great news! Glad it worked! I put some mitigations in the code so hopefully it won't happen to other people!

**Comment by alexbelgium** [comment]
> Happy new year :) I was already working on a fix that you merged which appeared to be empty... Will look into it again why that didn't work. Sorry! Will fix it with the next update.

Sorry I don't know how that happened! Seems I clicked on a wrong button while I was looking at it on github from my phone... Collaboration is something new to me on github I'll be more careful in the future and really appreciate your input, help to improve addons and pull requests. BTW I quickly made some commits to avoid erasing the config in /addons_config config, as the mv command overwrites by default, so I put a requisite in the if function that it moves the files only if the folder in /addons_config doesn't exist... We could also use mv -n to avoid overwrite, it would be cleaner also. 

And happy new year! :)

**Comment by bruvv** [comment]
Seems to me that I have auto merge on for the master branch on my repo.. haha that apparently force merges everything. didn't know that. Switched to sub branches and all should be good now with the latest merge :)

**Comment by almico** [comment]
Question: will a similar issue impact other *arrs? 😄

**Comment by bruvv** [comment]
No I have been running just fine with the other servarrs addons with the new config location :)

**Comment by almico** [comment]
> No I have been running just fine with the other servarrs addons with the new config location :)

I asked because I would have stated exactly the same for Jackett...
No more than 4 hours ago 😆
I did absolutely nothing after upgrading a few days ago, and it broke without my intervention.

**Comment by alexbelgium** [comment]
> > No I have been running just fine with the other servarrs addons with the new config location :)
> 
> I asked because I would have stated exactly the same for Jackett... No more than 4 hours ago 😆 I did absolutely nothing after upgrading a few days ago, and it broke without my intervention.

Don't you have auto update on? That might be a probable cause

**Comment by almico** [comment]
> Don't you have auto update on? That might be a probable cause

Most likely, I had it disabled, but... Who knows? 😆
